File: tetrahymena_part_A_me1528.R

#install.packages("tidyverse")
#install.packages("dplyr")
#install.packages("ggplot2")

#load dplyr package
library("dplyr")

#load the ggplot2 package
library(ggplot2)

#load the tidyverse packa
library("tidyverse")
library("readr")

#Step 1: load the script the tsv file
mydata <- read_tsv("tetrahymena.tsv")
View(mydata)

#Step 2: removing the rows which has diameter <= 19.2
mydata <- mydata[!mydata$diameter <= 19.2,]

#Step 3: computing the mean for culture and glucose grouping, respectively
meanCulture <- mydata[,2:4] %>% group_by(culture) %>% summarise_all(funs(mean(., na.rm=TRUE)))

meanGlucose <- mydata %>% group_by(glucose) %>% summarise_all(funs(mean(., na.rm=TRUE)))

#Step 4: adding new columns for log_concentration and log_diameter, respectively
mydata$log_conc <- log(mydata$conc)

mydata$log_diameter <- log(mydata$diameter)

#Step 5 and 6: doing scatter plot using ggplot and geom_smooth to display the smooth line
ggplot(mydata, aes(x=log_conc, y=log_diameter,shape=glucose,color=glucose)) + geom_point() + geom_smooth(method=lm, se=FALSE, fullrange=TRUE)

#save the plot into PDF
ggsave("tetrahymena_part_A_From_R_me1528.pdf")

File: tetrahymena_part_A_me1528.py
#!/usr/bin/python
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

#Step 1: reading the dataset 
mydata = pd.read_csv("tetrahymena.tsv",delimiter='\t')

#Step 2: Filtering out cells that have a diameter <= 19.2
mydata = mydata[mydata.diameter > 19.2]

#Step 3: Using mean concentration and diameter over technical replicates to remove psuedo-replicates 

meanCulture = mydata.groupby(mydata.culture).mean()
print(meanCulture)

meanGlucose = mydata.groupby(mydata.glucose)["conc","diameter"].mean()
print(meanGlucose)

#Step 4: Creating new columns for diameter and concentration that have a natural log of each, respectively  
mydata = mydata.assign(log_conc = np.log(mydata.conc))

mydata = mydata.assign(log_diameter = np.log(mydata.diameter))

#Step 5 
#Using sns package from seaborn to create scatter plot of log concentration vs log diameter

#Step 6 (BONUS): adding a line for seperation of data
sns_plot = sns.lmplot("log_conc","log_diameter",data=mydata,hue='glucose',legend=False,markers=["o","^"],fit_reg=True,scatter_kws={"s":50})

plt.legend(loc="upper right")

#Saving the output file as a PDF 
sns_plot.savefig("tetrahymena_part_A_From_Python_me1528.pdf")
